### PR TITLE
Adding kill functionallity for win.

### PR DESCRIPTION
**Pull request by Igmat**
Hotfix for #40.
Used implementation for `kill` function from [`node-cross-spawn-with-kill`](https://github.com/kentcdodds/cross-spawn-with-kill) so many thanks to @kentcdodds.

This fix will provide sync killing in `win32` environment and will use default node implementation for `*nix`.
## Motivation

Due to this [discussion](https://github.com/kulshekhar/ts-jest/issues/32#issuecomment-256791280) and that `kill` is part of [node API](https://nodejs.org/api/child_process.html#child_process_child_kill_signal) I think that it's not [out of scope](https://github.com/IndigoUnited/node-cross-spawn/issues/40#issuecomment-243608659) for this package.
## TODO:
- [x] Add tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ npm-debug.*
 test/fixtures/(*
 test/fixtures/shebang_noenv
 test/tmp
+.vscode/
```
Comment (kentcdodds) [inline .gitignore]: This should probably be in your [global gitignore](https://help.github.com/articles/ignoring-files/).

Comment (Igmat) [inline .gitignore]: You are right. I'll revert it in next commit.

**Comment by Igmat** [comment]
I've added tests, but they are useless as for now, because they don't fail when this `kill` implementation is removed. Seems, that my case [here](https://github.com/kulshekhar/ts-jest/issues/32#issuecomment-256755269) is much more complicated then just not killing child process and it failed only with some special circumstances that I can't figure out right now.

**Comment by satazor** [comment]
@Igmat the motivation is that, on windows, the `kill` option offered by NodeJS does not work?

**Comment by Igmat** [comment]
Partially. Actually in simple cases it works, and that's why those tests do not fail. But in more complicated case default `kill` works good on *nix systems and doesn't on Windows. As for now, I'm creating test that fully repeats my case and narrow it, to find out minimal circumstances when it will fail with default `kill`.

**Comment by satazor** [comment]
@Igmat Yes a reproducible test case would be awesome. I see many popular projects using `cross-spawn` and never heard of this problem. One of those projects is `execa` which uses `.kill()`.

**Comment by Igmat** [comment]
I've added pretty big complicated case and commented new `kill` implementation in order to show that it fails without it, and runs successfully with it. Now, I'll start to remove extra dependencies to narrow it to the most simple case.

**Comment by Igmat** [comment]
As I said it fails on Windows. See [this build](https://ci.appveyor.com/project/satazor/node-cross-spawn/build/252). Strange thing that it was succeeded for `node v0.12`.

**Comment by Igmat** [comment]
Ok, so as I understand problem is that I spawn `jest` in `watch` mode and it spawns something else and it prevents normal killing in Windows systems. So now, I'll try to investigate how actually jest's watch works and find out what happens with processes.

**Comment by Igmat** [comment]
@satazor, ok I've done my best to test this stuff. But as for now, I have no idea how to narrow it more... I'll probably ask guys from [`Jest`](https://github.com/facebook/jest) what actually happens when it called with `--watch/--watchAll` flags and come back to it later.
But here is a test for it, and you can easily check that this fix forces your package to act same on Windows and *nix systems. I guess that it is enough to accept this PR, is it?

**Comment by Igmat** [comment]
@kentcdodds may be you have some thoughts on it?

**Comment by kentcdodds** [comment]
If Jest's spawning is causing issues for you, you might try the `--runInBand` flag and see if that makes a difference.

**Comment by johanneswuerbach** [comment]
In testem I'm using similar code https://github.com/testem/testem/blob/f32d3edb38852d44fe842f0bceab132ca4ccbe6e/lib/process-ctl.js#L200-L220 as libuv under Windows always force kills processes (no SIGTERM) https://github.com/libuv/libuv/blob/912d42b7285fc29869472106d1c736f34ac91521/src/win/process.c#L1171, which in some cases causes child processes of the spawned process to never be killed.

The magic is the `\T` flag here.

I would also like to see this merged, but please with support for disabling `\F` somehow as GUI applications can actually be gracefully terminated under Windows, just terminal apps always require `\F`.

**Comment by satazor** [comment]
I'm preparing a new release and I'm looking at this PR again.

Regarding overwriting `kill`, how do you expect to support the [`signal`](https://nodejs.org/api/child_process.html#child_process_subprocess_kill_signal) to send?

**Comment by satazor** [comment]
I've landed a major overhaul to the project in the master branch.

For now, I will cloose this PR since there's no answer for my question regarding specifying signals with kill. I might reopen if necessary.

**Comment by Igmat** [comment]
It was too long ago. Actually, I'm not an expert in that how processes managed in Windows, so I can't answer for your previous question @satazor. I just wanted to have same behavior at different systems for CI purposes and all I need was just simple `kill()`.
Also, it seems that @johanneswuerbach knows much more about different flags and etc. If it still makes sense for you, I may rebase this branch.

**Comment by satazor** [comment]
Unfortunately, not having a cross platform `kill()` is a blocker for me. If that its improved to at least support the Windows signals, we can reopen this MR.

**Pull request closed without merging**